### PR TITLE
feat(Profile flow): Respond to/review an incoming CR

### LINKED ISSUE #13519: [Profile flow] Respond to/review an incoming CR
image

- "Ignore" just closes the popup

Figma: https://www.figma.com/file/ibJOTPlNtIxESwS96vJb06/%F0%9F%91%A4-Profile-%7C-Desktop?type=design&node-id=2460-200309&mode=design&t=qVMy2WcrL4fvxnwK-0

### PR DESCRIPTION
**Pull request by caybro**
### What does the PR do

- implement the new "Review contact request" popup & flow
- adjust storybook

Fixes #13519

### Affected areas

ProfileDialogView,ProfileContextMenu

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

image

### CODE REVIEW

File: ui/app/mainui/Popups.qml [hunk 1]
```diff
@@ -467,6 +479,23 @@ QtObject {
...
+            id: reviewContactRequestPopupComponent
+            ReviewContactRequestPopup {
+                onAccepted: {
+                    rootStore.contactStore.acceptContactRequest(publicKey, contactRequestId)
+                    Global.displaySuccessToastMessage(qsTr("Contact request accepted"))
+                    close()
+                }
+                onDiscarded: {
+                    // FIXME should call `rootStore.contactStore.dismissContactRequest(publicKey, contactRequestId)` here?
+                    Global.displaySuccessToastMessage(qsTr("Contact request ignored"))
```
Comment (caybro) [inline ui/app/mainui/Popups.qml:492]: One question here @MishkaRogachev: here we should just "ignore" the request but all the other popups (ActivityCenter, Settings/Contacts, etc) still call the `dismissContactRequest`; so what to do here?

Comment (MishkaRogachev) [reply ui/app/mainui/Popups.qml:492]: i'm not aware of latest requirements, but i can answer what `dismissContactRequest` does:
- mark provided or latest CR as _Dismissed_
- set contact's `ContactRequestLocalState` as ContactRequestStateDismissed. All subsequent CRs from this contact will not change the contact's status and trigger AC notifications

So my opinion is to use `dismissContactRequest` to _ignore_ a contact

**Review by alexjba — APPROVED** [review]
LGTM! Very nice storybook flow you've set-up there!
Minor questions inline

File: ui/app/mainui/Popups.qml [hunk 2]
```diff
@@ -467,6 +479,23 @@ QtObject {
             }
         },
 
+        Component {
+            id: reviewContactRequestPopupComponent
+            ReviewContactRequestPopup {
+                onAccepted: {
+                    rootStore.contactStore.acceptContactRequest(publicKey, contactRequestId)
+                    Global.displaySuccessToastMessage(qsTr("Contact request accepted"))
+                    close()
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/app/mainui/Popups.qml:488]: Hmm does it? I'll double check as I'm using it elsewhere too but the trick here is that this particular signal is overridden :) So it won't auto close it

File: ui/app/mainui/Popups.qml [hunk 3]
```diff
@@ -222,6 +223,17 @@ QtObject {
         openPopup(sendContactRequestPopupComponent, popupProperties, cb)
     }
 
+    function openReviewContactRequestPopup(publicKey, contactDetails, cb) {
+        try {
+            const crDetails = rootStore.contactStore.getLatestContactRequestForContactAsJson(publicKey)
+            if (crDetails.from !== publicKey)
+                console.warn("Popups.openReviewContactRequestPopup: not matching publicKey!!!")
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/app/mainui/Popups.qml:230]: Yup, will do

[the remark this answers is not in the document]
Comment (caybro) [reply ui/app/mainui/Popups.qml]: Done

File: ui/app/mainui/Popups.qml (hunk 2 above)
[the remark this answers is not in the document]
Comment (caybro) [reply ui/app/mainui/Popups.qml:490]: Checked: merely emitting the signal will not auto close the dialog, one needs to call `accept()` or `reject()`

File: ui/app/mainui/Popups.qml (hunk 1 above)
Comment (caybro) [reply ui/app/mainui/Popups.qml:494]: Thanks, done

**Review by alexandraB99 — APPROVED** [review]

**Pull request merged**